_episodes/11-esmvalcoreapi.md

```diff
@@ -0,0 +1,511 @@
+---
+title: "ESMValCore API in a Jupyter notebook"
+teaching: 20
+exercises: 30
+compatibility: ESMValTool, ESMValCore v2.11.0
+
+questions:
+- "How to find data for ESMValTool in a Jupyter Notebook?"
+- "How to use preprocessor functions?"
+objectives:
+- "Use the Dataset object"
+- "Import and use preprocessor functions"
+- "View and check the data"
+keypoints:
+- "API can be used as a helper to develop recipes"
+- "Preprocessors can be used in a Jupyter Notebook to check the output"
+- "Use `datasets_to_recipe` helper to start making recipes"
+---
+
+In this episode we will introduce the ESMValCore API in a jupyter notebook. This is reformatted 
+from material from this [blog post][easy-ipcc-blog]{:target="_blank"}
+by Peter Kalverla. There's also material from the 
+[example notebooks][docs-notebooks]{:target="_blank"} and the 
+[API reference documentation][api-reference]{:target="_blank"}.
+
+## Start JupyterLab
+A [jupyter notebook](https://jupyter.org/){:target="_blank"} is an interactive document where 
+you can run code.
+You will need to use a python environment with ESMValTool and ESMValCore installed.
+
+## Find Datasets with facets
+We have seen from running available recipes that ESMValTool is able to find data from facets that
+were given in the recipe. We can use this in a Notebook, including filling out the facets for 
+data definition. 
+To do this we will use the `Dataset` object from the API. Let's look at this example. 
+
```

Comment:
We usually provide guidance on where this code should go. For instance,  is this in a Notebook or a Python file or a YAML file? As it stands, it is not clear where this code will go and what it will be used for. Partly because we haven't discussed ESMValCore APIs in the tutorial.